scrapy/http/response/__init__.py

```diff
@@ -183,6 +205,8 @@ def follow(
         .. versionadded:: 2.0
            The *flags* parameter.
         """
+        if encoding is None:
```

Comment:
Allowing None in the signature because `TextResponse` needs it allowed in its method implementations and those call the base class methods (`TextResponse` methods don't call `Response.follow()` but I'm not sure if it can be detected statically).